WordPress/gutenberg
# Post publish upload media dialog: handle more block types

## What?

This PR adds new block types to the post publish upload media dialog, so that external images in them are also considered for upload.

As part of this work, it also deduplicates URLs and generates unique filenames before upload.

This PR fixes #64899.

## Why?

The post publish upload media dialog suggests uploading external media to the media library, for both performance and reliability reasons. However, it currently only does so for Image blocks, which limits its usefulness.

## How?

Custom logic was added to handle `media-text` and `cover` blocks, as well as to generate adequate filenames before uploading.

## Testing Instructions





1. Clear your media library
2. Start a new post in the editor
3. Paste the below HTML into it

<details>
<summary>Post contents</summary>

```html

<figure class="wp-block-image size-large is-resized"><img src="https://s.w.org/style/images/about/WordPress-logotype-standard.png" alt="WordPress Logotype" style="width:127px;height:auto"/></figure>



<div class="wp-block-cover"><span aria-hidden="true" class="wp-block-cover__background has-background-dim"></span><img class="wp-block-cover__image-background" alt="WordPress Logotype in a cover block" src="https://s.w.org/style/images/about/WordPress-logotype-standard.png" data-object-fit="cover"/><div class="wp-block-cover__inner-container">
<p class="has-text-align-center has-large-font-size">Text</p>
</div></div>



<div class="wp-block-media-text is-stacked-on-mobile"><figure class="wp-block-media-text__media"><img src="https://s.w.org/style/images/about/WordPress-logotype-standard.png" alt="WordPress Logotype in a media-text block"/></figure><div class="wp-block-media-text__content">
<p>Text</p>
</div></div>



<figure class="wp-block-image size-large is-resized"><img src="https://s.w.org/style/images/about/WordPress-logotype-standard.png?forceDifferentUrl" alt="A separate copy of the WordPress Logotype, with a different URL" style="width:127px;height:auto"/></figure>

```

</details>

4. Hit Publish
5. Ensure that there are four images in the upload list, and that they all have the correct alt text (which will be different for each of them)
    <img width="281" alt="image" src="https://github.com/user-attachments/assets/cbf35cc9-d7ef-4ea6-ab9a-e1db75916566">
6. Hit Upload
7. Ensure that the post visually looks the same
8. Switch to the code editor, and ensure that all the URLs now point to uploaded media (somewhere in `/wp-content/uploads/`)
9. Consult your media library
10. Ensure exactly two images were uploaded, and that they're named `WordPress-logotype-standard.png` and `WordPress-logotype-standard-<some UUID>.png`. The second file is expected because we used a different URL (via a query parameter) on the last block, and it's expected to have the UUID for deduplication of the filename.

### Testing Instructions for Keyboard
N/A, no UI changes.
